wgsl/index.bs

```diff
@@ -3526,6 +3686,35 @@ Issue: Which index is used when it's out of bounds?
            (OpAccessChain)
 </table>
 
+<table class='data'>
+  <caption>Getting a pointer to an array element from a pointer to an array</caption>
+  <thead>
+    <tr><th>Precondition<th>Conclusion<th>Description
+  </thead>
+  <tr algorithm="sized array indexed pointer selection">
+       <td class="nowrap">
+          |p| : ptr&lt;|SC|,array&lt;|T|,|N|&gt;&gt;<br>
+          |i| : *Int*
+       <td class="nowrap">
+           |p|[|i|] : ptr&lt;|SC|,|T|&gt;
+       <td>Compute a pointer to the |i|'<sup>th</sup> element of the array referenced by the pointer |p|.<br>
+           If |i| is outside the range [0,|N|-1], then an index in the range [0, |N|-1] is used instead.<br>
+           The [=originating variable=] of the resulting pointer is
+           the same as the originating variable of |p|.<br>
+           (OpAccessChain)
+  <tr algorithm="array indexed pointer selection">
+       <td>|p| : ptr&lt;|SC|,array&lt;|T|&gt;&gt;<br>
+          |i| : *Int*
+       <td class="nowrap">
+           |p|[|i|] : ptr&lt;|SC|,|T|&gt;
+       <td>Compute a pointer to the |i|'<sup>th</sup> element of the runtime-sized array referenced by the pointer |p|.<br>
+           If at runtime the array has |N| elements, and |i| is outside the range [0,|N|-1], then an index in the
```

Comment:
I don't think we can describe it this way. Currently, WebGPU group wants to allow many ways of handling out of bounds accesses:
  - discard the whole draw call
  - behave like RBA, which means reading can return anything within the buffer (not within the array), or zeroes, or other stuff
  - behave like D3D, which means reading always returns zeroes, and writing is ignored

Reply:
This is the same logic as has already landed for indexing into an array value and a reference-to-array.
So this objection should be pulled out into a new issue, I think.

Reply:
Following up in #1649